src/com/sfuentes/MedianAndSmallest.java

```diff
@@ -0,0 +1,24 @@
+package com.sfuentes;
+
+public class MedianAndSmallest {
+
+  public double calculateSmallest(double[] grades) {
+    double smallest = grades[0];
+    for (int i = 0; i < grades.length - 1; i++) {
```

Comment:
This method fails if the last number in the array is the smallest, check your condition in the for loop.